src/commands/branch.rs

```diff
@@ -61,7 +61,14 @@ pub struct BranchDeleteArgs {
 /// preceded by a "+". For information about branches, see
 /// https://github.com/martinvonz/jj/blob/main/docs/branches.md.
 #[derive(clap::Args, Clone, Debug)]
-pub struct BranchListArgs;
+pub struct BranchListArgs {
+    /// Show branches which local/remote targets are in the given revisions.
+    ///
+    /// If any of local/remote targets matched, the branch (including
+    /// non-matching targets) will be listed.
+    #[arg(long, short)]
+    revisions: Option<RevisionArg>,
```

Comment:
(Optional, unsure about this): I find the `revisions` name a bit confusing. To me, it suggests listing branches that *include* a revision (so, I'd expect `--revision X` to do what `--revision X:` actually does).

One option is to rename the arg to `--at-revision`. But that would break the analogy with `jj git push`.

Reply:
Like `git tag --contains`?

I don't think `-r` is accurate, but I don't have a better option. `--at-revision` sounds like tags were version controlled and lists all tags at that revision.

Reply:
I haven't had any better ideas than keeping `--revisions`. Feel free to resolve this whenever you give up on the quest for a better name; `--revisions` is probably fine.